src/braket/circuits/circuit.py

```diff
@@ -891,6 +892,24 @@ def copy(self) -> Circuit:
         """
         return self._copy()
 
+    def decompose(self) -> Circuit:
+        """
+        Call a decomposition pass across the circuit.
+
+        Returns:
+            Circuit: Circuit decomposed by one level.
```

Comment:
Is the recursive decomposition you gave the same as the one as in here: https://github.com/aws/amazon-braket-examples/blob/main/examples/advanced_circuits_algorithms/QFT/QFT.ipynb

Reply:
Yes, I think it is the same one.

The point is not recursive, but give users decompose levels. In your demo, you demoed:
```
circ.decompose().decompose()
```
A better UI might be `circ.decompose()` have a default level 1, which means your example becomes
```
circ.decompose(2)
```
And you will have something like
```
circ.decompose(level="infinite")
```
for recursive QFT.

Then for the choice of recursive QFT and your current QFT, you can either have
```
# Recursive QFT
circuit.qft_recursive()
# Your current QFT
circuit.qft()
```
Or
```
# Recursive QFT
circuit.qft(method="recursive")
# Your current QFT
circuit.qft()
```

In addition, users might want to just decompose certain Composite Operators, then it will be nice to have
```
circ.decompose(operators=["qft"], level="infinite")
```